ui/hooks/usePolling.ts

```diff
@@ -0,0 +1,49 @@
+import { useEffect, useRef } from 'react';
+
+type UsePollingOptions = {
+  callback?: (pollingToken: string) => (pollingToken: string) => void;
+  startPollingByNetworkClientId: (
+    networkClientId: string,
+    options: any,
+  ) => Promise<string>;
+  stopPollingByPollingToken: (pollingToken: string) => void;
+  networkClientId: string;
+  options?: any;
+};
+
+const usePolling = (usePollingOptions: UsePollingOptions) => {
+  const pollTokenRef = useRef<null | string>(null);
+  const cleanupRef = useRef<null | ((pollingToken: string) => void)>(null);
+  let isMounted = false;
+  useEffect(() => {
+    isMounted = true;
+
+    const cleanup = () => {
+      if (pollTokenRef.current) {
+        usePollingOptions.stopPollingByPollingToken(pollTokenRef.current);
+        cleanupRef.current?.(pollTokenRef.current);
+      }
+    };
+    // Start polling when the component mounts
+    usePollingOptions
+      .startPollingByNetworkClientId(
+        usePollingOptions.networkClientId,
+        usePollingOptions.options,
+      )
+      .then((pollToken) => {
+        pollTokenRef.current = pollToken;
+        cleanupRef.current = usePollingOptions.callback?.(pollToken) || null;
+        if (!isMounted) {
+          cleanup();
+        }
+      });
+
+    // Return a cleanup function to stop polling when the component unmounts
+    return () => {
+      isMounted = false;
+      cleanup();
+    };
+  }, [usePollingOptions.networkClientId, usePollingOptions.options]);
```

Comment:
Worried about the options object in the dependencies. I'm still not convinced that we shouldn't stringify it. We don't care if the reference changes, only if the content itself changes